src/main/java/com/wellsmith/sapien/persistence/avro/AvroConfig.java

```
@@ -0,0 +1,44 @@
+package com.wellsmith.sapien.persistence.avro;
+
+import org.apache.avro.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+public class AvroConfig {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroConfig.class);
+
+    @Bean
+    public Map<String, Schema> avroSchemaCache() {
+        Map<String, Schema> schemaMap = new HashMap<>();
+        try {
+            File avroFileDirectory = new ClassPathResource("sapien-schemas").getFile();
+            File[] files = avroFileDirectory.listFiles((d, s) -> s.toLowerCase().endsWith(".avsc"));
+            if (files != null) {
+                for (File file : files) {
+                    try {
+                        Schema schema = new Schema.Parser().parse(file);
+                        schemaMap.putIfAbsent(schema.getFullName(), schema);
+                        System.out.println("Cached schema " + schema.getFullName() + ": " + schema);
+                    } catch (IOException ie) {
+                        LOG.error("Could not read Avro schema file with name {}", file.getName());
```

let's allow exceptions to propagate. unless there is a good reason to allow it to fail quietly.

---

If an Exception is thrown and not handled downstream, the Flink workflow will just stop working, so at some point you've got to catch the Exception and log it.  Presumable at some point we'll be sending out a Kafka message to be consumed by a service that not only stores it somewhere, but alerts someone about the Exception/Error.

Also, this will be replaced with a call to a schema repository in the (hopefully) near future.

---

yeah, i was guessing if the schemas fail to load here, then the workflow will not work anyway. if that's true then i would err on the side of failing fast. then later when we have fancier exception publishing, we could be more tolerant at the point of failure.

---

Hopefully we'll end up with a schema registry, but you're right: if the schemas can't be read or the schema registry can't be reached, the workflow should fail fast.  I need to implement having the Exceptions being thrown up to the highest level, then dealt with there (to being with by failing fast and logging the stack trace).  That way when we come up with a more elegant solution, only one substitution will have to be made.

---

Fixed this instance.